File: Scadex.Model/Dtos/Camera/Queries/CameraCaptureDto.cs
using Scadex.Core.Model;
using static Scadex.Model.Enums.EntityEnums;

namespace Scadex.Model.Dtos.Camera.Queries;

public class CameraCaptureDto : IDto
{
    public long Id { get; set; }
    public Guid CameraId { get; set; }
    public CaptureType Type { get; set; }
    public CaptureStatus Status { get; set; }
    public DateTime CapturedAtUtc { get; set; }
    public int? DurationSec { get; set; }
    public string? RelativePath { get; set; }
    public long? SizeBytes { get; set; }
    public string? FailureReason { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public Guid? RequestedByUserId { get; set; }
}

File: Scadex.WebUI/src/models/camera/queries/cameraCaptureDto.ts
/**
 * Ayna: Scadex.Model/Dtos/Camera/Queries/CameraCaptureDto.cs
 * Sözleşme: docs/api-contract/11-camera.md
 */
import type { CaptureStatus, CaptureType } from '@/models/enums/entityEnums';

export interface CameraCaptureDto {
  /** IDENTITY PK — `number`, `string` değil. Guid olmadığına dikkat. */
  id: number;
  cameraId: string;

  type: CaptureType;
  status: CaptureStatus;

  /**
   * Görüntünün ANI (ISO-8601 UTC) — satırın oluşturulma zamanı değil.
   * Klipte kaydın fiilen başladığı andır.
   */
  capturedAtUtc: string;

  /** Klip süresi (saniye); anlık görüntüde `null`. */
  durationSec: number | null;

  /**
   * Dosyanın sunucu köküne göre GÖRELİ yolu (örn.
   * `uploads/captures/2026/08/31/{guid}.jpg`).
   *
   * Tam URL DEĞİLDİR — `captureFileUrl()` ile `API_BASE_URL`'e eklenir.
   * `Pending` ve `Failed` iken `null`.
   */
  relativePath: string | null;

  sizeBytes: number | null;

  /** `Failed` ise sebep. Başarısız çekim de bir satır bırakır. */
  failureReason: string | null;

  /** Saklama süresinin sonu; `null` ise süresiz. */
  expiresAt: string | null;

  requestedByUserId: string | null;
}
